src/goImpl.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+'use strict';
+
+import vscode = require('vscode');
+import cp = require('child_process');
+import { getBinPath } from './util';
+import { promptForMissingTool } from './goInstallTools';
+
+interface GoImplInput {
+	receiver: string;
+	interface: string;
+}
+
+// Supports only passing interface, see TODO in implCursor to finish
+const inputRegex = /^(\w+\ \*?\w+\ )?([\w.]+)$/;
+
+export function implCursor() {
+	let cursor = vscode.window.activeTextEditor.selection;
+	return vscode.window.showInputBox({
+		placeHolder: 'f *File io.Closer',
+		prompt: 'What interface do you want to implement?'
+	}).then(implInput => {
+		if (typeof implInput === 'undefined') {
+			return;
+		}
+		const matches = implInput.match(inputRegex);
+		if (!matches) {
+			vscode.window.showInformationMessage(`Not parsable input: ${implInput}`);
+			return;
+		}
+
+		// TODO: automatically detect type name at cursor
```

Comment:
I was thinking of right after the type. So it would need some detection to find the first empty line after the type.